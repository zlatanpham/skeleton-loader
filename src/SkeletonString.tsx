import * as React from 'react';
import styled, { CSSObject } from 'styled-components';
import {
  SkeletonThemeConsumer,
  SkeletonThemeProps,
} from './SkeletonThemeProvider';
import {
  defaultBaseColor,
  defaultHighlightColor,
  defaultDuration,
  getStyles,
} from './shared';

const TextWrapper = styled.span<
  SkeletonStringDefaultProps & { theme: SkeletonThemeProps }
>`
  ${props => getStyles(props)}
`;

interface SkeletonStringDefaultProps {
  duration: number;
  baseColor: string;
  highlightColor: string;
  customStyle?: CSSObject | TemplateStringsArray;
}
export interface SkeletonStringProps {
  duration?: number;
  baseColor?: string;
  highlightColor?: string;
  customStyle?: CSSObject | TemplateStringsArray;
}

export class SkeletonString extends React.Component<SkeletonStringProps> {
  static defaultProps: SkeletonStringDefaultProps = {
    baseColor: defaultBaseColor,
    highlightColor: defaultHighlightColor,
    duration: defaultDuration,
  };

  render() {
    return (
      <SkeletonThemeConsumer>
        {theme => (
          <TextWrapper
            {...this.props as SkeletonStringDefaultProps}
            theme={theme}
          >
            {this.props.children}
          </TextWrapper>
        )}
      </SkeletonThemeConsumer>
    );
  }
}
