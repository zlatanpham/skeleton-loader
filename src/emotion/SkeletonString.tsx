import * as React from 'react';
import { CSSObject, SerializedStyles } from '@emotion/core';
import styled from '@emotion/styled';
import {
  SkeletonThemeConsumer,
  SkeletonThemeProps
} from './SkeletonThemeProvider';
import {
  defaultBaseColor,
  defaultHighlightColor,
  defaultDuration,
  getStyles
} from './shared';

const TextWrapper = styled.span<
  SkeletonStringDefaultProps & { theme: SkeletonThemeProps }
>`
  ${props => getStyles(props)}
`;

interface SkeletonStringDefaultProps {
  duration: number;
  baseColor: string;
  highlightColor: string;
  customStyle?: CSSObject | SerializedStyles;
}
export interface SkeletonStringProps {
  duration?: number;
  baseColor?: string;
  highlightColor?: string;
  customStyle?: CSSObject | SerializedStyles;
}

export class SkeletonString extends React.Component<SkeletonStringProps> {
  static defaultProps: SkeletonStringDefaultProps = {
    baseColor: defaultBaseColor,
    highlightColor: defaultHighlightColor,
    duration: defaultDuration
  };

  shouldComponentUpdate() {
    return false;
  }

  render() {
    return (
      <SkeletonThemeConsumer>
        {theme => (
          <TextWrapper
            {...this.props as SkeletonStringDefaultProps}
            theme={theme}
          >
            {this.props.children}
          </TextWrapper>
        )}
      </SkeletonThemeConsumer>
    );
  }
}
