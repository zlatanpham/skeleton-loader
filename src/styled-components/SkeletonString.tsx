import * as React from 'react';
import styled, { CSSObject } from 'styled-components';
import {
  SkeletonThemeConsumer,
  SkeletonThemeProps
} from './SkeletonThemeProvider';
import {
  defaultBaseColor,
  defaultHighlightColor,
  defaultDuration,
  getStyles
} from './shared';

const TextWrapper = styled.span<
  SkeletonStringDefaultProps & { theme: SkeletonThemeProps }
>`
  ${props => getStyles(props)}
`;

interface SkeletonStringDefaultProps
  extends React.HTMLAttributes<HTMLSpanElement> {
  duration: number;
  baseColor: string;
  highlightColor: string;
  customStyle?: CSSObject | TemplateStringsArray;
}
export interface SkeletonStringProps
  extends React.HTMLAttributes<HTMLSpanElement> {
  duration?: number;
  baseColor?: string;
  highlightColor?: string;
  customStyle?: CSSObject | TemplateStringsArray;
}

export const SkeletonString: React.FC<SkeletonStringProps> = React.memo(
  ({
    baseColor = defaultBaseColor,
    highlightColor = defaultHighlightColor,
    duration = defaultDuration,
    children,
    ...rest
  }) => (
    <SkeletonThemeConsumer>
      {theme => (
        <TextWrapper
          {...({
            ...rest,
            baseColor,
            highlightColor,
            duration
          } as SkeletonStringDefaultProps)}
          theme={theme}
        >
          {children}
        </TextWrapper>
      )}
    </SkeletonThemeConsumer>
  ),
  () => true
);
