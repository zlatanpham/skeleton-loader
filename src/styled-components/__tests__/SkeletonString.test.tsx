import 'jest-styled-components';
import * as React from 'react';
import { render } from '@testing-library/react';
import { SkeletonString } from '../SkeletonString';

describe('<SkeletonString />', () => {
  it('Should match snapshot', () => {
    const { container } = render(
      <SkeletonString>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque
        eget ipsum sodales, tincidunt tortor at, feugiat quam. Morbi dignissim
        ligula nisi, sit amet tempus eros porta at.
      </SkeletonString>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
