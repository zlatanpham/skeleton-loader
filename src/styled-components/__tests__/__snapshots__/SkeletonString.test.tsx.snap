// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkeletonString /> Should match snapshot 1`] = `
.c0 {
  background-color: #eeeeee;
  background-image: linear-gradient( 90deg, #eeeeee, #f5f5f5, #eeeeee );
  background-size: 200px 100%;
  background-repeat: no-repeat;
  display: inline-block;
  -webkit-animation: ceFrVu 1.5s ease-in-out infinite;
  animation: ceFrVu 1.5s ease-in-out infinite;
  color: transparent !important;
}

.c0 > * {
  color: transparent !important;
}

<span
  class="c0"
>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque eget ipsum sodales, tincidunt tortor at, feugiat quam. Morbi dignissim ligula nisi, sit amet tempus eros porta at.
</span>
`;
